src/ghostfragment/fragmenting/inter_finder.hpp

```diff
@@ -0,0 +1,34 @@
+#include <chemist/molecule/fragmented_molecule.hpp>
+#include <chemist/nucleus/fragmented_nuclei.hpp>
+
+#include <algorithm>
+#include <cmath>
+#include <cstdlib>
+#include <map>
+#include <utility>
+
+namespace ghostfragment::fragmenting {
+
+using mol_type  = chemist::FragmentedMolecule;
+using nuke_type = chemist::FragmentedNuclei;
+
+using frag_type   = std::vector<std::size_t>;
+using system_type = std::vector<frag_type>;
+
+using group_type = std::vector<std::size_t>;
+
+using inter_type = std::vector<std::size_t>;
+using res_type   = std::map<inter_type, float>;
+
+using log_type = std::map<std::pair<std::size_t, std::size_t>, inter_type>;
+
+void find_group_intersections(const system_type& fragments, inter_type inter,
+                              std::size_t recent_add, res_type& m,
+                              log_type& log, group_type& group);
+
+void find_pair_intersections(const system_type& fragments, log_type& log);
+
+void create_system(const mol_type& mol, system_type& fragments);
```

Comment:
These should be unit tested. Also I'd recommend making these functions pure (in particular actually return the returns, don't modify the inputs).

Reply:
Alternatively, you can avoid the need to unit test these functions if you avoid declaring these functions in the header file (by making them live in the source, they become implementation details).

Reply:
I had read somewhere that modifying the input was better for optimizing the time. Though, if we already established that it is fast enough, I doubt this is will be the thing that slows it down. 

And I'll move the functions to the source. I can't imagine any other use for them beyond the intersection finder, so it makes sense to only live in the source.

Reply:
I'm not sure what your source was, but I think that is true for C and maybe C++ in the early 90s. Modern C++ guarantees return-value optimization in some cases, and compilers exploit it automatically in many other cases.

Reply:
I know the feeling :rofl: